rfc/001_constraintsV2.md

```diff
@@ -0,0 +1,172 @@
+in one place, to speed up the work needed on GHC upgrades and package
+relaeses with big impact. This will make the format more complex than
+what V1 is, but apart from the initial work needed I don't think
+that's much of an issue. I would prefer to not have to do this... but
+I think it's necessary to keep people sane.
+
+EDIT: Another idea which I would prefer. Extend `curator` with some
+helper commands to do these. E.g. `curator disable <package>
+<comment>`. I'd need to test how the decode/encode round trip behaves,
+I don't want it to arbitrarily reorder everything all of the time.
+
+### Range shorthand
+We can still allow `- package < range` for bounds as well as `range:`
+if no other properties are set. I'd lean towards not allowing it as
+it's more syntax. As is, removing the range doesn't require you to
+remove the `name:` prefix, and you can't remove the `name:` prefix if
+other properties are present (e.g. disabled tests)
+
+### Maintainer configuration
+We could add another level of nesting under the maintainer name to
+allow new maintainer-level features, one thing that has been requested
+is to be able to do something like what's below. We can add this later
+without breaking the format (if there's a list => only packages, if
+object => there should be a packages property with that list).
+
+```yaml
+packages:
+  "Bob":
+    dont-notify-when-a-reverse-dependency-has-bounds-issues-with-my-packages: true
+    packages:
+      - package1
+      - package2
+```
+
+### Stricter validation
+
+V1 has the feature/issue that multiple maintainers can list the same
+package. I'm not a fan of this as a `< 0 ` may be added in only one
+place and a maintainer may think the package is included when it's
+not. I propose we error out of this occurs (exception: Don't do this
+if packages are listed in the V1/ghc upgrade sections)
+
+### Minor things
+
+These are not introduced in V2, and I think doing them later would be fine.
+
+* `range: < 0` is a hack, it might be clearer to have `disabled:
+  true`? OTOH `range: < 0` sticks out more visually.
+* `tell-me-when-its-released` is rarely used, but it would make more
+  sense if its format was a range condition for triggering it rather
+  than "notify me when a newer version than this is available".
+* Require a github handle for users. Now and then we run across the
+  issue that we don't notify because there is no github handle. I'm
+  guessing the missing handles are just accidental. I think it would
+  make sense to require this as we usually don't end up sending
+  e-mails. We could make the parsing of the maintainer string stricter
+  or add it under the proposed "Maintainer configuration". I'd lean
+  towards stricter parsing to not have to add that level of
+  "indentation" to every maintainer.
+* Along that line, should we remove the e-mail addresses as we seldom
+  make use of them? Might still be good to have on occasion.
```

Comment:
I think it would be good to remove these as privacy self-conscious people might not upload to Stackage because of the email being so easily to grab